src/main/java/graphs/GraphClass.java

```diff
@@ -43,6 +43,22 @@ public void addNode(GraphNode node) {
 
     public void addEdge(GraphEdge edge) {
         edges.add(edge);
+        if(! isDirected()) {
+            duplicateAsUndirectedEdge(edge);
+        }
+    }
+
+    private void duplicateAsUndirectedEdge(GraphEdge edge) {
+        edges.add(new GraphEdge(edge.getRightNode(), edge.getLeftNode(), edge.getWeight()));
+    }
+
+    public GraphClass duplicateUndirectedEdges() {
```

Comment:
do we need this? is there any other way to add an edge besides the constructor(s) and addEdge?